### PR TITLE
Split impl at 'for' if a line break is needed

### LINKED ISSUE #1037: Split impl<...> X for Y<...> at 'for'
screenshot from 2016-06-05 18-52-08

### PR DESCRIPTION
**Pull request by dawirstejeck**
Fix #1037.

### CODE REVIEW

File: tests/source/impls.rs [hunk 1]
```diff
@@ -13,20 +13,20 @@ pub impl Foo for Bar {
 }
 
 pub unsafe impl<'a, 'b, X, Y: Foo<Bar>> !Foo<'a, X> for Bar<'b, Y> where X: Foo<'a, Z> {
-    fn foo() { "hi" }    
+    fn foo() { "hi" }
```
Comment (dawirstejeck) [inline tests/source/impls.rs]: Oops, my vim config removed these trailing white spaces... Have they been there intentionally?

Comment (nrc) [inline tests/source/impls.rs]: yes

File: tests/source/impls.rs [hunk 2]
```diff
@@ -100,3 +100,7 @@ mod m {
 
 impl<BorrowType, K, V, NodeType, HandleType> Handle<NodeRef<BorrowType, K, V, NodeType>, HandleType> {
 }
+
+impl<BorrowType, K, V, NodeType, HandleType> PartialEq
+    for Handle<NodeRef<BorrowType, K, V, NodeType>, HandleType> {
+}
```
Comment (nrc) [inline tests/source/impls.rs]: could you change the source of the test to be incorrectly formatted please

**Comment by nrc** [comment]
looks good, just a couple of minor things to fix, thanks!

**Comment by dawirstejeck** [comment]
@nrc done. Maybe you can stop some of those integration test (except the last one)? Had some trouble with restoring that white space and was to quick with pushing my commits :laughing:.

**Comment by nrc** [comment]
Thank you!

**Comment by dawirstejeck** [comment]
My pleasure.

**Pull request merged**